drivers: firmware: tisci: Add secure path support

Kronosblaster opened the pull request:
**Summary**
Add support for secure TISCI communication paths to enable usage of secure threads for communication.

**Changes:**
- Add is_secure flag to tisci_config for secure/non-secure path selection
- Implement tisci_msg_requires_secure_path() to identify messages requiring secure path
- Add secure header wrapping for outbound messages on secure path
- Add secure header parsing for inbound messages on secure path
- Improve error handling with proper semaphore cleanup in error paths
- Add ti,is-secure device tree property to indicate secure host entities

**Testing**
Change the host-id and thread ids in the core specific dt file and run the /tests/drivers/tisci

soumya-TI commented on drivers/firmware/tisci/tisci.c:
```diff
@@ -58,16 +59,45 @@ struct tisci_data {
+ * Board configuration, security, and OTP messages must use the secure path.
+ * All other messages use the non-secure path by default.
+ *
+ * Reference: MCU+SDK sciclient.c Sciclient_getCurrentContext()
```
This is no need I guess, referencing a different code base with no proper access.

soumya-TI commented on drivers/firmware/tisci/tisci.c:
```diff
@@ -58,16 +59,45 @@ struct tisci_data {
 	struct tisci_data *data = dev->data;
+	const struct tisci_config *config = dev->config;
+
+	if (!config->is_secure && tisci_msg_requires_secure_path(msg_type)) {
```
Why are we having a block here, not to send a message on a non-secure path, its job of server to nack it. isn't it? Client driver should be dumb enough to send the message, if that so, then all the messages should go as its is there shouldn't be a need to differentiate secure and non-secure messages. If we are differentiating we are building a intelligence, that can be added to support both secure and non-secure pathways on the same driver. Isn't it?

glneo replied:
TISCI firmware not responding will result in a timeout waiting for the result which is treated as an error the same as a reject message.

Kronosblaster committed:
drivers: firmware: tisci: Reorganise macros

- Add missing TISCI message type macros in tisci.h
- Clean macro organisation
- Add Secure indicator for message that require secure thread

Signed-off-by: Dave Joseph <d-joseph@ti.com>

Kronosblaster committed:
drivers: firmware: tisci: Add secure property

- Add ti,is-secure device tree property to indicate secure host entities
- Add is_secure field in device config

Signed-off-by: Dave Joseph <d-joseph@ti.com>

Kronosblaster committed:
drivers: firmware: tisci: Add secure path support

Add support for secure TISCI communication paths to enable
message exchange with secure contexts on TI SoCs.
- Add secure header wrapping for outbound messages on secure path
- Add secure header parsing for inbound messages on secure path

Signed-off-by: Dave Joseph <d-joseph@ti.com>

Kronosblaster committed:
drivers: firmware: tisci: Fix semaphore cleanup

- Add proper semaphore cleanup in error paths where it was missing

Signed-off-by: Dave Joseph <d-joseph@ti.com>

glneo commented on drivers/firmware/tisci/tisci.h:
```diff
@@ -21,24 +21,56 @@
+#define TISCI_MSG_ENTER_SLEEP 0x0301
 
-/* Clock requests */
+/* Security Messages */
```
These are still all out of order, why keep updating this series without first addressing my simple request to keep the definitions in numerical order?

Kronosblaster replied:
This is updated. Aside from TISCI_MSG_OPEN_DEBUG_FWLS and TISCI_MSG_SA2UL_GET_DKEK which are grouped by function all other messages have been numerically ordered
